maven-java/kalix-maven-plugin/src/main/java/kalix/DeployMojo.java

```diff
@@ -36,32 +35,51 @@ public class DeployMojo extends AbstractMojo {
     @Parameter(property = "kalixContext")
     private String kalixContext;
 
+    @Parameter(property = "kalixProject")
+    private String kalixProject;
+
     @Parameter(defaultValue = "30000", property = "cliTimeoutMs", required = true)
     private Long cliTimeoutMs;
 
     private final Log log = getLog();
 
     /**
-     * We deploy by invoking the services deploy command
+     * We deploy by invoking the services deploy command only when `deployToKalix` is `true`
+     * and the current Kalix project matches with 'kalixCurrentProject'.
      */
     public void execute() throws MojoExecutionException {
-        log.info("Deploying project to Kalix");
+       final List<String> commandLine;
+       final int deploymentResult = 0;
+       if (kalixProject.isEmpty()){
+           log.info("The variable `kalixProject` hasn't been set. Therefore, not deploying to Kalix");
```

Comment:
One thought now that I have slept on it for a night, should we perhaps adhere to the kalixProject setting when present in maven pom but just deploy to current project if not set instead?

That would make close to zero effort for newcomers/trying it out (they'll already have selected a project) but still allow more advanced users to control which Kalix project a maven project should deploy to.

Reply:
It didn't cross my mind the `timestamp`. 👍 I'll have a look. 
Regards the optionality of `kalixProject`, sounds good. I think we need then to make sure through documentation that users are aware of the option and that is a recommended feature. So they don't miss it altogether.

Reply:
I thought about if we can perhaps log something saying that
"kalixProject not set, deploying to default project configured through kalix cli"

But maybe that is a bit too noisy to have on each deploy?